index.bs

```diff
@@ -535,167 +738,319 @@ To determine whether an |attribute| <dfn>contains a javascript: URL</dfn>:
+</div>
 
-NOTE: [=SanitizerConfig/Remove=] matches on name and namespace, so adding an
-    element with attributes would still remove the matching element from the
-    {{SanitizerConfig/removeElements}} and {{SanitizerConfig/replaceWithChildrenElements}} lists.
+1. Set |element| to the result of [=canonicalize a sanitizer element with attributes=] with
+    |element|.
+1. [=SanitizerConfig/Remove=] |element| from
+    |configuration|["{{SanitizerConfig/replaceWithChildrenElements}}"].
+1. If |configuration|["{{SanitizerConfig/elements}}"] [=map/exists=]:
+    1. If |configuration|["{{SanitizerConfig/elements}}"] does not [=map/contain=] |element|:
+        1. [=Comment=]: This is the case with a global allow-list that does not yet contain
+            |element|.
+        1. [=list/Append=] |element| to |configuration|["{{SanitizerConfig/elements}}"].
+        1. Return true.
+    1. [=Comment=]: This is the case with a global allow-list that already contains |element|.
+        We'll need to do quite a bit of extra work to maintain our validity critertia in case
+        per-element attributes are used.
+    1. Let |original element| be a copy of |element|.
+    1. If |element|["{{SanitizerElementNamespaceWithAttributes/attributes}}"] [=map/exists=]:
+        1. If |configuration|["{{SanitizerConfig/attributes}}"] [=map/exists=]:
+            1. Set |element|["{{SanitizerElementNamespaceWithAttributes/attributes}}"] to the
+                [=set/difference=] of
+                |element|["{{SanitizerElementNamespaceWithAttributes/attributes}}"] and
+                |configuration|["{{SanitizerConfig/attributes}}"].
+        1. If |configuration|["{{SanitizerConfig/removeAttributes}}"] [=map/exists=]:
+            1. Set |element|["{{SanitizerElementNamespaceWithAttributes/attributes}}"] to the
+                [=set/intersection=] of
+                |element|["{{SanitizerElementNamespaceWithAttributes/attributes}}"] and
+                |configuration|["{{SanitizerConfig/removeAttributes}}"].
+    1. If |element|["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"] [=map/exists=]:
+        1. If |configuration|["{{SanitizerConfig/attributes}}"] [=map/exists=]:
+            1. Set |element|["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"] to the
+                [=set/difference=] of
+                |element|["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"] and
+                |configuration|["{{SanitizerConfig/attributes}}"].
+        1. If |configuration|["{{SanitizerConfig/removeAttributes}}"] [=map/exists=]:
+            1. Set |element|["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"] to the
+                [=set/difference=] of
+                |element|["{{SanitizerElementNamespaceWithAttributes/removeAttributes}}"] and
+                |configuration|["{{SanitizerConfig/removeAttributes}}"].
+    1. If |element| equals |original element|, then return false.
```

Comment:
I think it's obvious what you are doing here, but we need to make sure this a defined operation.